packages/core/admin/_internal/node/createBuildContext.ts

```diff
@@ -170,11 +170,14 @@ const createBuildContext = async ({
 
   const features = strapiInstance.config.get('features', undefined);
 
+  const serveAdminPanel = strapiInstance.config.get('admin.serveAdminPanel', true);
+
   const { bundler = 'webpack', ...restOptions } = options;
 
   const buildContext = {
     appDir,
-    basePath: `${adminPath}/`,
+    // If serveAdminPanel is false, use relative paths for separate deployment
+    basePath: serveAdminPanel ? adminPath : '',
```

Comment:
```suggestion
    basePath: serveAdminPanel ? `${adminPath}/` : '',
```

This is the change that works for me. In both cases (true and false) of `serveAdminPanel`